### PR TITLE
Implement data grouping functionality

### PR DESCRIPTION
**Pull request by roubkar**
- Implement data grouping method (by color, stroke-style/dasharray, chart index)
- Use data grouping during data processing
- Add a helper function for getting all object nested paths

### CODE REVIEW

File: aim/web/ui_v2/src/pages/Metrics/components/TableColumns/TableColumns.tsx [hunk 1]
```diff
@@ -52,7 +54,14 @@ function getTableColumns(): ITableColumn[] {
       title: 'Iteration',
       width: 150,
     },
-  ];
+  ].concat(
+    paramColumns.map((param) => ({
+      dataKey: param.replaceAll('.', '_'),
```
Comment (gorarakelyan) [inline aim/web/ui_v2/src/pages/Metrics/components/TableColumns/TableColumns.tsx:59]: why are dots replaced here?

Comment (roubkar) [reply aim/web/ui_v2/src/pages/Metrics/components/TableColumns/TableColumns.tsx:59]: There was a problem with getting value by nested path inside `BaseTable` component, so as a quick fix I decided to pass flatten object as row data

**Review by VkoHov — APPROVED** [review]

File: aim/web/ui_v2/src/pages/Metrics/components/TableColumns/TableColumns.tsx (hunk 1 above)
Comment (roubkar) [reply aim/web/ui_v2/src/pages/Metrics/components/TableColumns/TableColumns.tsx]: The issue is actually resolved with the latest commit

**Review by KaroMourad — APPROVED** [review]
lgtm

**Comment by roubkar** [comment]
@gorarakelyan this PR is ready for review

**Comment by gorarakelyan** [comment]
Awesome, will check it out

**Review by gorarakelyan — APPROVED** [review]
lgtm

**Pull request merged**